ui/src/app/workflows/components/workflow-dag/workflow-dag.tsx

```diff
@@ -433,26 +489,28 @@ export class WorkflowDag extends React.Component<WorkflowDagProps, WorkflowDagRe
                 });
             });
         });
+        this.graph.edges = edges.filter(e => this.graph.nodes.has(e.v) && this.graph.nodes.has(e.w)).map(e => this.generateEdge(e));
+    }
+
+    private generateEdge(edge: Edge) {
         // `h` and `v` move the arrow heads to next to the node, otherwise they would be behind it
         const h = this.state.horizontal ? this.nodeSize / 2 : 0;
         const v = !this.state.horizontal ? this.nodeSize / 2 : 0;
-        this.graph.edges = edges
-            .filter(e => this.graph.nodes.has(e.v) && this.graph.nodes.has(e.w))
```

Comment:
does this removal change behaviour? introduce bugs?

Reply:
This is not removed. It is present here https://github.com/argoproj/argo/pull/3967/files#diff-f3e947c640962abeb48e9dd7aabdf668R492